packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -771,7 +772,28 @@ class _AdbLogReader extends DeviceLogReader {
   // we default to true in case none of the log lines match
   bool _acceptedLastLine = true;
 
+  // The format of the line is controlled by the '-v' parameter passed to
+  // adb logcat. We are currently passing 'time', which has the format:
+  // mm-dd hh:mm:ss.milliseconds Priority/Tag( PID): ....
   void _onLine(String line) {
+    final Match timeMatch = AndroidDevice._timeRegExp.firstMatch(line);
+    if (timeMatch == null) {
+      return;
+    }
+    if (_timeOrigin != null) {
+      final String timestamp = timeMatch.group(0);
+      DateTime time = _adbTimestampToDateTime(timestamp);
+      if (time.isBefore(_timeOrigin)) {
+        // Ignore log messages before the origin.
+        printTrace('skipped old log line: $line');
```

Comment:
I'm just being consistent with the existing behaviour: log messages that are ignored are still printed via printTrace. See the bottom of this function.